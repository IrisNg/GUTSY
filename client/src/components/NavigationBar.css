.brand, .navigation-bar__main-nav .main-nav__primary .main-nav__brand {
  color: #f45800;
  font-family: 'Pacifico', cursive;
  margin: 0; }

.container, .navigation-bar__main-nav {
  margin: 0 3rem;
  padding: .8rem 1rem; }

.navigation-bar__main-nav {
  display: flex;
  justify-content: space-between;
  align-items: center; }
  .navigation-bar__main-nav .main-nav__primary {
    display: flex;
    align-items: center;
    width: 36%; }
    .navigation-bar__main-nav .main-nav__primary .main-nav__brand {
      font-size: 2.3rem; }
    .navigation-bar__main-nav .main-nav__primary .main-nav__search {
      height: 2.2rem;
      width: 100%;
      display: flex;
      align-items: center;
      margin-left: 1.5rem; }
      .navigation-bar__main-nav .main-nav__primary .main-nav__search .main-nav__search-input {
        height: 100%;
        width: 100%;
        padding: 0 .8rem;
        border: 1px solid #8d8d8d;
        border-top-left-radius: 3px;
        border-bottom-left-radius: 3px;
        font-family: inherit;
        background-color: #f9f9f9;
        box-shadow: 2.5px 2.5px 2.5px inset #e9e9e9;
        outline: none; }
        .navigation-bar__main-nav .main-nav__primary .main-nav__search .main-nav__search-input:hover, .navigation-bar__main-nav .main-nav__primary .main-nav__search .main-nav__search-input:focus {
          box-shadow: none;
          background-color: white;
          border-color: black; }
      .navigation-bar__main-nav .main-nav__primary .main-nav__search .main-nav__search-button {
        background-color: #2f2f2f;
        color: white;
        height: 100%;
        font-weight: 700;
        padding: 0 .8rem;
        font-size: .85rem;
        border-top-right-radius: 3px;
        border-bottom-right-radius: 3px; }
        .navigation-bar__main-nav .main-nav__primary .main-nav__search .main-nav__search-button:hover {
          background-color: #494949; }
  .navigation-bar__main-nav .main-nav__items {
    display: flex;
    justify-content: flex-end;
    align-items: center; }
    .navigation-bar__main-nav .main-nav__items li {
      margin: 0 .7rem;
      font-size: .95rem; }
    .navigation-bar__main-nav .main-nav__items .main-nav__sign-in {
      border: 1px solid #595959;
      padding: .5rem;
      box-sizing: border-box;
      border-radius: 3px; }
    .navigation-bar__main-nav .main-nav__items .main-nav__basket,
    .navigation-bar__main-nav .main-nav__items .main-nav__discover {
      display: flex;
      align-self: flex-end;
      flex-direction: column;
      align-items: center;
      font-size: .75rem;
      border-left: 1px solid #c7c7c7;
      padding: 0 .7rem;
      margin-right: 0;
      color: #8d8d8d;
      font-weight: 400; }
      .navigation-bar__main-nav .main-nav__items .main-nav__basket i,
      .navigation-bar__main-nav .main-nav__items .main-nav__discover i {
        font-size: 1.4rem;
        margin-bottom: .2rem; }
    .navigation-bar__main-nav .main-nav__items .main-nav__basket {
      margin-left: 0; }

.navigation-bar__categories {
  display: flex;
  justify-content: space-around;
  margin: 0;
  border-top: 1px solid #c3c3c3;
  border-bottom: 1px solid #c3c3c3;
  font-size: .85rem;
  padding: .9rem 4rem; }

/*# sourceMappingURL=NavigationBar.css.map */
