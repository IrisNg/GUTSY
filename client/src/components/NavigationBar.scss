@import "_Default";
.navigation-bar__main-nav {
    @extend %container;
    display: flex;
    justify-content: space-between;
    align-items: center;
    .main-nav__primary {
        display: flex;
        align-items: center;
        width: 36%;
        .main-nav__brand {
            @extend %brand;
            font-size: 2.3rem;
            cursor: pointer;

        }
        .main-nav__search {
            @extend %attached-field;
            width: 100%;
         
            margin-left: 1.5rem;
            .main-nav__search-input {
                @extend %attached-input;
                
            }
            .main-nav__search-button {
                @extend %attached-button;
                
            }
        }
    }
    .main-nav__items {
        display: flex;
        justify-content: flex-end;
        align-items: center;
        li {
            margin: 0 .7rem;
            font-size: .95rem;
            cursor: pointer;

        }
        .main-nav__sell,
        .main-nav__register {
            &:hover {
                text-decoration: underline;
            }
        }
        .main-nav__sign-in {
            border: 1px solid #595959;
            padding: .5rem;
            box-sizing: border-box;
            border-radius: 3px;
            &:hover {
                background-color: #404040;
                color: white;
                border-color: transparent;
            }
        }
        .main-nav__basket,
        .main-nav__discover {
            display: flex;
            align-self: flex-end;
            flex-direction: column;
            align-items: center;
            font-size: .75rem;
            border-left: 1px solid #c7c7c7;
            padding: 0 .7rem;
            margin-right: 0;
            color: #8d8d8d;
            font-weight: 400;
            i {
                font-size: 1.4rem;
                margin-bottom: .2rem;
            }
            &:hover i {
                color: #303030;
            }
        }
        .main-nav__basket {
            margin-left: 0;
        }
    }
}

.navigation-bar__categories {
    display: flex;
    justify-content: space-between;
    margin: 0;
    border-top: 1px solid #c3c3c3;
    border-bottom: 1px solid #c3c3c3;
    font-size: .85rem;
    padding: .9rem 5rem;
    li {
        cursor: pointer;

    }
}