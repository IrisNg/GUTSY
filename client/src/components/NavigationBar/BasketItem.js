import React from 'react';

const BasketItem = ({ item }) => {
   return (
      <div className="basket-item">
         <img src={item.productImage} alt={`basket item ${item.productName}`} className="basket-item__image" />
         <div className="basket-item__product-details">
            <h5 className="basket-item__name">{item.productName}</h5>
            <h3 className="basket-item__price">$ {item.productPrice}</h3>
         </div>
      </div>
   );
};

export default BasketItem;
