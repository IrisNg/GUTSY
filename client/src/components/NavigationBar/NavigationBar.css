.navigation-bar__main-nav .main-nav__brand {
  color: #f45800;
  font-family: 'Pacifico', cursive;
  margin: 0; }

.navigation-bar__main-nav {
  margin: 0 5rem;
  padding: 0.6rem 1rem; }

.navigation-bar__main-nav .search-bar__input, .navigation-bar__expanded-categories {
  border: 1px solid #c6c6c6; }

.navigation-bar__main-nav .main-nav__sell:hover, .navigation-bar__expanded-categories .expanded-categories__category span:hover, .navigation-bar__expanded-categories .expanded-categories__sub-category:hover {
  text-decoration: underline; }

.navigation-bar__main-nav .search-bar {
  height: 2.2rem;
  display: flex;
  align-items: center; }

.navigation-bar__main-nav .search-bar__input {
  padding: 0 0.8rem;
  border-top-left-radius: 3px;
  border-bottom-left-radius: 3px;
  font-family: inherit;
  background-color: #f9f9f9;
  box-shadow: 2px 2px 2px inset #e9e9e9;
  outline: none;
  height: 100%;
  width: 100%;
  letter-spacing: 0.04rem; }
  .navigation-bar__main-nav .search-bar__input:hover, .navigation-bar__main-nav .search-bar__input:focus {
    box-shadow: none;
    background-color: white;
    border-color: black; }

.navigation-bar__main-nav .search-bar__button {
  background-color: #2f2f2f;
  color: white;
  font-weight: 700;
  letter-spacing: 0.03rem;
  padding: 0 0.8rem;
  font-size: 0.85rem;
  border-top-right-radius: 3px;
  border-bottom-right-radius: 3px;
  height: 100%; }
  .navigation-bar__main-nav .search-bar__button:hover {
    background-color: #494949; }

.navigation-bar__main-nav {
  display: flex;
  justify-content: space-between;
  align-items: center; }
  .navigation-bar__main-nav .main-nav__primary {
    display: flex;
    align-items: center;
    width: 36%; }
  .navigation-bar__main-nav .main-nav__brand {
    font-size: 2.3rem;
    cursor: pointer; }
  .navigation-bar__main-nav .search-bar {
    width: 100%;
    margin-left: 1.5rem; }
  .navigation-bar__main-nav .main-nav__items {
    display: flex;
    justify-content: flex-end;
    align-items: center; }
    .navigation-bar__main-nav .main-nav__items li.main-nav__sell,
    .navigation-bar__main-nav .main-nav__items li.main-nav__auth,
    .navigation-bar__main-nav .main-nav__items li.main-nav__user {
      margin: 0 0.7rem;
      font-size: 0.95rem; }
  .navigation-bar__main-nav .main-nav__sell {
    cursor: pointer; }
  .navigation-bar__main-nav .main-nav__user {
    display: flex;
    align-items: center;
    cursor: default; }
  .navigation-bar__main-nav .main-nav__user-image {
    border-radius: 50%;
    width: 30px;
    height: 30px;
    margin-right: 3px; }
  .navigation-bar__main-nav .main-nav__auth {
    border: 1px solid #595959;
    padding: 0.45rem 0.5rem;
    box-sizing: border-box;
    border-radius: 3px;
    display: flex;
    align-items: center;
    cursor: pointer; }
    .navigation-bar__main-nav .main-nav__auth i {
      margin-right: 5px;
      font-size: 1.2rem; }
    .navigation-bar__main-nav .main-nav__auth:hover {
      background-color: #404040;
      color: white;
      border-color: transparent; }
  .navigation-bar__main-nav .main-nav__basket,
  .navigation-bar__main-nav .main-nav__discover {
    display: flex;
    align-self: flex-end;
    flex-direction: column;
    align-items: center;
    font-size: 0.75rem;
    border-left: 1px solid #c7c7c7;
    padding: 0 0.7rem;
    color: #8d8d8d;
    font-weight: 400;
    cursor: pointer; }
    .navigation-bar__main-nav .main-nav__basket i,
    .navigation-bar__main-nav .main-nav__discover i {
      font-size: 1.4rem;
      margin-bottom: 0.2rem; }
    .navigation-bar__main-nav .main-nav__basket:hover i,
    .navigation-bar__main-nav .main-nav__discover:hover i {
      color: #303030; }
  .navigation-bar__main-nav .main-nav__discover {
    margin-left: 0.7rem; }
  .navigation-bar__main-nav .main-nav__basket {
    padding-right: 0; }

.navigation-bar__categories {
  border-top: 1px solid #c6c6c6; }

.navigation-bar__categories--inactive {
  border-top: 1.5px solid #e0e0e0; }

.navigation-bar__categories--inactive .navigation-bar__main-categories {
  display: none; }

.navigation-bar__main-categories {
  display: flex;
  justify-content: space-between;
  margin: 0;
  border-bottom: 1px solid #c6c6c6;
  font-size: 0.9rem;
  padding: 0 5rem; }
  .navigation-bar__main-categories li {
    cursor: pointer;
    padding: 0.7rem;
    text-align: center;
    display: inline;
    border-bottom: 3px solid transparent; }
    .navigation-bar__main-categories li.main-category--active {
      border-bottom: 3px solid #f45800; }

.navigation-bar__expanded-container {
  position: relative; }

.navigation-bar__expanded-categories {
  padding-left: 0;
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  background-color: white;
  margin: 0 5rem;
  width: calc(100vw - 11.25rem);
  display: flex;
  justify-content: space-around;
  border-top: none;
  padding: 2rem 3rem;
  min-height: 8rem;
  font-size: 0.93rem;
  letter-spacing: 0.03rem; }
  .navigation-bar__expanded-categories .expanded-categories__category {
    font-weight: 700; }
    .navigation-bar__expanded-categories .expanded-categories__category span {
      cursor: pointer;
      color: #5f5f5f; }
  .navigation-bar__expanded-categories .expanded-categories__sub-categories {
    padding: 0.4rem 0 0 0.3rem;
    font-weight: 400;
    font-size: 0.91rem; }
  .navigation-bar__expanded-categories .expanded-categories__sub-category {
    margin: 0.5rem 0;
    cursor: pointer; }

.navigation-bar__expanded-categories--inactive {
  display: none; }

/*# sourceMappingURL=NavigationBar.css.map */
