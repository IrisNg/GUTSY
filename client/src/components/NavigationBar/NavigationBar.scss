@import '_Default';
.navigation-bar__main-nav {
    @extend %container;
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.main-nav__primary {
    width: 36%;
    display: flex;
    align-items: center;
}

.main-nav__brand {
    @extend %brand;
    font-size: 2.3rem;
    cursor: pointer;
}

.search-bar {
    @extend %attached-field;
    width: 100%;
    margin-left: 1.5rem;
}

.search-bar__input {
    @extend %attached-input;
}

.search-bar__button {
    @extend %attached-button;
}

.main-nav__items {
    display: flex;
    justify-content: flex-end;
    align-items: center;
    li.main-nav__sell,
    li.main-nav__auth .google-auth,
    li.main-nav__user {
        margin: 0 0.7rem;
        font-size: 0.95rem;
    }
}

.main-nav__sell {
    @extend %hover-underline;
    cursor: pointer;
}

.main-nav__user {
    display: flex;
    align-items: center;
    cursor: default;
}

.main-nav__user-image {
    width: 30px;
    height: 30px;
    margin-right: 3px;
    border-radius: 50%;
}

.main-nav__auth .google-auth {
    padding: 0.45rem 0.5rem;
    display: flex;
    align-items: center;
    border: 1px solid #595959;
    border-radius: 3px;
    box-sizing: border-box;
    cursor: pointer;
    i {
        margin-right: 5px;
        font-size: 1.2rem;
    }
    &:hover {
        background-color: #404040;
        color: white;
        border-color: transparent;
    }
}

.main-nav__basket,
.main-nav__discover {
    align-self: flex-end;
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 0 0.7rem;
    font-size: 0.75rem;
    border-left: 1px solid #c7c7c7;
    color: #8d8d8d;
    font-weight: 400;
    cursor: pointer;
    i {
        margin-bottom: 0.2rem;
        font-size: 1.4rem;
    }
    &:hover i {
        color: #303030;
    }
}

.main-nav__discover {
    margin-left: 0.7rem;
}

.main-nav__basket {
    padding-right: 0;
    position: relative;
    i {
        position: relative;
    }
    &:hover .basket__list-container {
        visibility: visible;
    }
}

.basket__counter {
    position: absolute;
    top: -0.8rem;
    right: -0.8rem;
    width: 0.8rem;
    height: 0.8rem;
    padding: 10px;
    display: flex;
    align-items: center;
    justify-content: center;
    border-radius: 50%;
    background-color: $logo-color;
    color: white;
    font-size: 0.8rem;
    font-family: 'PT Sans', sans-serif;
    letter-spacing: -0.04rem;
}

.basket__list-container {
    visibility: hidden;
    position: absolute;
    top: 2rem;
    right: -1rem;
    z-index: 2;
    padding-top: 1.2rem;
    &:hover {
        visibility: visible;
    }
}

.basket__list {
    @extend %border;
    padding: 0.5rem 1rem 0 1rem;
    background-color: white;
    color: map-get($font-color, normal);
    border-radius: 4px;
    cursor: default;
}

.basket__empty-message {
    width: max-content;
    display: block;
    margin-top: 1rem;
    margin-bottom: 1.5rem;
    text-align: center;
}

.basket-item {
    width: fit-content;
    min-width: 230px;
    padding: 1rem 0;
    display: flex;
    justify-content: space-between;
    text-align: right;
    border-bottom: 1px solid $border-color;
}

.basket-item__image {
    width: 90px;
    height: 70px;
    object-fit: cover;
    border-radius: 4px;
}

.basket-item__product-details {
    align-self: stretch;
    padding: 0.2rem;
}

.basket-item__name {
    margin-bottom: 0.3rem;
    font-size: 0.9rem;
    font-weight: 400;
}

.basket-item__price {
    font-size: 0.95rem;
}

.basket__total {
    margin-top: .4rem;
    padding: 0 .2rem;
    display: flex;
    justify-content: space-between;
}

.basket__total-header {
    font-size: 1.1rem;
}

.basket__total-value {
    font-size: 1.1rem;
}

.basket__checkout-button {
    @extend %button;
    width: 100%;
    margin: 1.2rem 0;
    background-color: $logo-color;
    &:hover {
        background-color: $logo-color;
    }
}

.navigation-bar__categories {
    border-top: 1px solid $border-color;
}

.navigation-bar__categories--inactive {
    border-top: 1.5px solid $border-color-light;
}

.navigation-bar__categories--inactive .navigation-bar__main-categories {
    display: none;
}

.navigation-bar__main-categories {
    margin: 0;
    padding: 0 5rem;
    display: flex;
    justify-content: space-between;
    border-bottom: 1px solid $border-color;
    font-size: 0.9rem;
    li {
        padding: 0.7rem;
        display: inline;
        text-align: center;
        border-bottom: 3px solid transparent;
        cursor: pointer;
        &.main-category--active {
            border-bottom: 3px solid $logo-color;
        }
    }
}

.navigation-bar__expanded-container {
    position: relative;
}

.navigation-bar__expanded-categories {
    width: calc(100vw - 11.25rem);
    min-height: 8rem;
    position: absolute;
    top: 0;
    left: 0;
    z-index: 1;
    margin: 0 5rem;
    padding: 2rem 3rem;
    display: flex;
    justify-content: space-around;
    @extend %border;
    border-top: none;
    background-color: white;
    font-size: 0.93rem;
    letter-spacing: 0.03rem;
}

.expanded-categories__category {
    font-weight: 700;
    span {
        cursor: pointer;
        color: map-get($font-color, light);
        @extend %hover-underline;
    }
}

.expanded-categories__sub-categories {
    padding: 0.4rem 0 0 0.3rem;
    font-weight: 400;
    font-size: 0.91rem;
}

.expanded-categories__sub-category {
    margin: 0.5rem 0;
    cursor: pointer;
    @extend %hover-underline;
}

.navigation-bar__expanded-categories--inactive {
    display: none;
}