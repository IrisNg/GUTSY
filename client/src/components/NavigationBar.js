import React from 'react';

import './NavigationBar.css';

class NavigationBar extends React.Component {
   render() {
      return (
         <div className="navigation-bar">
            <div className="navigation-bar__main-nav">
               <div className="main-nav__primary">
                  <h1 className="main-nav__brand">Gutsy</h1>
                  <div className="main-nav__search">
                     <input type="text" className="main-nav__search-input" placeholder="Search for items or shops" />
                     <button className="main-nav__search-button">Search</button>
                  </div>
               </div>
               <ul className="main-nav__items">
                  <li>Sell on Gutsy</li>
                  <li>Register</li>
                  <li className="main-nav__sign-in">
                     <button>Sign in</button>
                  </li>
                  <li className="main-nav__discover">
                     <i className="fas fa-toolbox" />
                     <span>Discover</span>
                  </li>
                  <li className="main-nav__basket">
                     <i className="fas fa-shopping-basket" />
                     <span>Basket</span>
                  </li>
               </ul>
            </div>
            <ul className="navigation-bar__categories">
               <li>Jewellery & Accessories</li>
               <li>Clothing & Shoes</li>
               <li>Home & Living</li>
               <li>Wedding & Party</li>
               <li>Toys & Entertainment</li>
               <li>Art & Collectibles</li>
               <li>Vintage</li>
            </ul>
         </div>
      );
   }
}
export default NavigationBar;
