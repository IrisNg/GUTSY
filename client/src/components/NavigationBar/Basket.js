import React from 'react';
import { connect } from 'react-redux';
import BasketItem from './BasketItem';

export class Basket extends React.Component {
   renderBasketCounter(basketItems) {
      if (basketItems.length === 0) {
         return null;
      }
      return <span className="basket__counter">{basketItems.length}</span>;
   }
   renderBasketItems(basketItems) {
      if (basketItems.length === 0) {
         return <span className="basket__empty-message">No items in here.</span>;
      }
      return basketItems.map(item => <BasketItem item={item} key={`basket ${item._id}`} />);
   }
   renderBasketTotal(basketItems) {
      if (basketItems.length === 0) {
         return null;
      }
      const basketTotal = basketItems.reduce((acc, cur) => acc + parseInt(cur.productPrice, 10), 0);
      return (
         <div className="basket__total">
            <h5 className="basket__total-header">Total</h5>
            <h6 className="basket__total-value">${basketTotal.toFixed(2)}</h6>
         </div>
      );
   }
   renderCheckoutButton(basketItems) {
      if (basketItems.length === 0) {
         return null;
      }
      return <button className="basket__checkout-button">Checkout</button>;
   }
   render() {
      const { basketItems } = this.props;
      return (
         <>
            <i className="fas fa-shopping-basket">{this.renderBasketCounter(basketItems)}</i>
            <span>Basket</span>
            <div className="basket__list-container">
               <div className="basket__list">
                  {this.renderBasketItems(basketItems)}
                  {this.renderBasketTotal(basketItems)}
                  {this.renderCheckoutButton(basketItems)}
               </div>
            </div>
         </>
      );
   }
}
const mapStateToProps = state => {
   return { basketItems: state.basket };
};
export default connect(mapStateToProps)(Basket);
