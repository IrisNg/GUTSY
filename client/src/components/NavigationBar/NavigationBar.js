import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import SearchBar from './SearchBar';
import Basket from './Basket';
import history from '../../history';
import productCategories from '../../productCategories';
import GoogleOAuth from '../GoogleOAuth/GoogleOAuth';
import flatProductCategories from '../../flatProductCategories';

import './NavigationBar.css';

class NavigationBar extends React.Component {
   state = {
      activeCategory: null,
      url: history.location.pathname
   };
   //Listen to changes in history, this affects if categories section is showing
   componentDidMount() {
      history.listen(location => {
         this.setState({ url: location.pathname });
      });
   }
   renderUser(userDetails) {
      return userDetails.username ? (
         <li className="main-nav__user">
            <img src={userDetails.userImage} className="main-nav__user-image" alt="user avatar from google" />
            {userDetails.username}
         </li>
      ) : null;
   }
   onDiscoverClick() {
      //Generate a random index number
      const randomIndex = Math.floor(Math.random() * (flatProductCategories.length - 1));
      //Find a category param based on the random Index
      const randomParam = flatProductCategories[randomIndex].param;
      //Navigate user to the random Category page
      history.push(`/category/${randomParam}`);
   }
   renderMainCategories() {
      return productCategories.map(mainCategory => {
         const { name, param } = mainCategory;
         return (
            <li
               onMouseOver={() => this.onCategoryHover(name)}
               className={this.handleMainCategoryStyle(name)}
               key={param}
               onClick={() => this.onCategoryClick(param)}
            >
               {name}
            </li>
         );
      });
   }
   // Show more product categories on hover
   onCategoryHover = activeCategory => {
      const categoryData = productCategories.find(mainCategory => mainCategory.name === activeCategory);
      this.setState({ activeCategory: categoryData });
   };
   onCategoryClick = param => {
      this.setState({ activeCategory: null });
      history.push(`/category/${param}`);
   };
   renderCategories() {
      if (!this.state.activeCategory) {
         return null;
      }
      return this.state.activeCategory.categories.map(category => {
         const { name, param, subCategories } = category;
         return (
            <li key={param} className="expanded-categories__category">
               <span onClick={() => this.onCategoryClick(param)}>{name}</span>
               <ul className="expanded-categories__sub-categories">{this.renderSubCategories(subCategories)}</ul>
            </li>
         );
      });
   }
   renderSubCategories(subCategories) {
      if (!subCategories) {
         return null;
      }
      return subCategories.map(subCategory => {
         const { name, param } = subCategory;
         return (
            <li key={param || 'null'} className="expanded-categories__sub-category" onClick={() => this.onCategoryClick(param)}>
               {name}
            </li>
         );
      });
   }
   handleCategoriesDisplay() {
      if (this.state.url === '/shop/') {
         return '--inactive';
      }
      return '';
   }
   handleExpandedCategoriesDisplay() {
      if (!this.state.activeCategory || this.state.activeCategory.categories.length === 0) {
         return '--inactive';
      }
      return '';
   }
   handleMainCategoryStyle(category) {
      const { activeCategory } = this.state;
      if (activeCategory && category === activeCategory.name) {
         return 'main-category--active';
      }
      return '';
   }
   render() {
      const { isSignedIn, userDetails } = this.props.auth;
      return (
         <div className="navigation-bar">
            {/* Main Navigation */}
            <div className="navigation-bar__main-nav">
               <div className="main-nav__primary">
                  {/* Logo */}
                  <h1 className="main-nav__brand">
                     <Link to="/">Gutsy</Link>
                  </h1>
                  {/* Search bar */}
                  <SearchBar />
               </div>
               {/* Buttons */}
               <ul className="main-nav__items">
                  <li className="main-nav__sell">
                     <Link to="/shop/create">Sell on Gutsy</Link>
                  </li>
                  {this.renderUser(userDetails)}
                  <li className="main-nav__auth">
                     <i className="fab fa-google-plus-g" />
                     <GoogleOAuth isSignedIn={isSignedIn} />
                  </li>
                  <li className="main-nav__discover" onClick={this.onDiscoverClick}>
                     <i className="fas fa-toolbox" />
                     <span>Discover</span>
                  </li>
                  <li className="main-nav__basket">
                     <Basket />
                  </li>
               </ul>
            </div>
            {/* Product Categories */}
            <div className={`navigation-bar__categories${this.handleCategoriesDisplay()}`} onMouseLeave={() => this.setState({ activeCategory: '' })}>
               <ul className="navigation-bar__main-categories">{this.renderMainCategories()}</ul>
               <div className="navigation-bar__expanded-container">
                  <ul className={`navigation-bar__expanded-categories${this.handleExpandedCategoriesDisplay()}`}>{this.renderCategories()}</ul>
               </div>
            </div>
         </div>
      );
   }
}
const mapStateToProps = state => {
   return { auth: state.auth };
};
export default connect(mapStateToProps)(NavigationBar);
