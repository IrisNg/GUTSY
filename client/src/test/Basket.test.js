import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { Basket } from '../components/NavigationBar/Basket';

describe('Basket', () => {
   const basketItems = [];
   const wrapper = shallow(<Basket basketItems={basketItems} />);
   it('should not display basket counter when empty', () => {
      expect(wrapper.instance().renderBasketCounter(basketItems)).toBeNull();
   });
   it('should not display basket total when empty', () => {
      expect(wrapper.instance().renderBasketTotal(basketItems)).toBeNull();
   });
   it('should display message when empty', () => {
      expect(wrapper.exists('span.basket__empty-message')).toBe(true);
   });
   it('should not display checkout button when empty', () => {
      expect(wrapper.instance().renderCheckoutButton(basketItems)).toBeNull();
   });
});

describe('Basket', () => {
   const basketItems = [
      { _id: 'a351', productPrice: '23.20', productName: 'hello kitty' },
      { _id: 'e9f9g', productPrice: '66.66', productName: 'nom nom' },
      { _id: 's9gnke', productPrice: '75.2', productName: 'beep beep' }
   ];
   it('matches the snapshot when it contains items', () => {
      const tree = renderer.create(<Basket basketItems={basketItems} />).toJSON();
      expect(tree).toMatchSnapshot();
   });
   it('displays 3 thumbnails, one per BasketItem', () => {
      const wrapper = mount(<Basket basketItems={basketItems} />);
      expect(wrapper.find('.basket-item img')).toHaveLength(3);
   });
});
