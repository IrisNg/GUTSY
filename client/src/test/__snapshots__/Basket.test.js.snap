// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basket matches the snapshot when it contains items 1`] = `
Array [
  <i
    className="fas fa-shopping-basket"
  >
    <span
      className="basket__counter"
    >
      3
    </span>
  </i>,
  <span>
    Basket
  </span>,
  <div
    className="basket__list-container"
  >
    <div
      className="basket__list"
    >
      <div
        className="basket-item"
      >
        <img
          alt="basket item hello kitty"
          className="basket-item__image"
        />
        <div
          className="basket-item__product-details"
        >
          <h5
            className="basket-item__name"
          >
            hello kitty
          </h5>
          <h3
            className="basket-item__price"
          >
            $ 
            23.20
          </h3>
        </div>
      </div>
      <div
        className="basket-item"
      >
        <img
          alt="basket item nom nom"
          className="basket-item__image"
        />
        <div
          className="basket-item__product-details"
        >
          <h5
            className="basket-item__name"
          >
            nom nom
          </h5>
          <h3
            className="basket-item__price"
          >
            $ 
            66.66
          </h3>
        </div>
      </div>
      <div
        className="basket-item"
      >
        <img
          alt="basket item beep beep"
          className="basket-item__image"
        />
        <div
          className="basket-item__product-details"
        >
          <h5
            className="basket-item__name"
          >
            beep beep
          </h5>
          <h3
            className="basket-item__price"
          >
            $ 
            75.2
          </h3>
        </div>
      </div>
      <div
        className="basket__total"
      >
        <h5
          className="basket__total-header"
        >
          Total
        </h5>
        <h6
          className="basket__total-value"
        >
          $
          164.00
        </h6>
      </div>
      <button
        className="basket__checkout-button"
      >
        Checkout
      </button>
    </div>
  </div>,
]
`;
